### PR TITLE
Styling of page headings with the same class, fixes #683

### LINKED ISSUE #683: Streamline Page Headings
### Describe the Bug
So far, we have multiple different heading-styles in our system.

This should be changed to a normal `<h2>` tag styled with css.
grafik
All headings should be the same as the one for the pages menu (see screenshot).

### PR DESCRIPTION
**Pull request by Wiedemsa**
### Short description

The headline styling in some templates was changed to match the proposed styling
Some missing headings have been added
Minor changes to the layout of some headings to have consistent styling

### Resolved issues

Fixes: #683

### CODE REVIEW

**Comment by Wiedemsa** [comment]
Seems like i worked with an outdated version...  therefore the conflict in <src/cms/templates/dashboard/dashboard.html>

**Review by timoludwig — COMMENTED** [review]
Thanks! :rocket: 
The class `font-normal` doesn't seem to do much - the font is already normal by default. I think you can just remove it.
And I found a few headings which you forgot - could you update them as well? Thanks!
- [statistics](https://github.com/Integreat/integreat-cms/blob/7fbd22f11cc29deedd1173dfe77355e453d77b85/src/cms/templates/statistics/statistics_overview.html#L10) (wrong class)
- [region users](https://github.com/Integreat/integreat-cms/blob/enhancement/steamline-page-headings/src/cms/templates/users/region/list.html) (heading completely missing)
- [push notifications](https://github.com/Integreat/integreat-cms/blob/enhancement/steamline-page-headings/src/cms/templates/push_notifications/push_notification_list.html) (heading completely missing)

**Comment by Wiedemsa** [comment]
Would it make sense to get rid of the mb-2 (=margin-bottom: 0.5rem ?) class to make the template even cleaner?
I have not yet tried but would it be possible to do the following:

In File:  integreat-cms/src/frontend/css/style.scss
from:
`h2.heading {
  font-size: 1.65rem;
}`
to:
`h2.heading {
  font-size: 1.65rem;
  margin-bottom: 0.5rem;
}`

**Comment by timoludwig** [comment]
> Would it make sense to get rid of the mb-2 (=margin-bottom: 0.5rem ?) class to make the template even cleaner?

Yes, this sounds like a good idea!
You can even use tailwind classes in the style-sheet (and since there is no larger font than these headings, you could as well convert them to `<h1>` if you want):
```
h1.heading {
  @apply text-3xl mb-2;
}
```

File: src/cms/templates/pages/page_tree.html [hunk 1]
```diff
@@ -129,6 +115,14 @@ <h2 class="heading font-normal mb-2">{% trans 'Pages' %}</h2>
...
             <input id="bulk-action-execute" type="submit" class="inline-block bg-gray-500 cursor-not-allowed h-full text-white font-bold py-2 px-4 ml-2 rounded" value="{% trans 'Execute' %}" />
         </div>
+        <div class="flex flex-wrap w-1/2 justify-end">
+            <a href="{% url 'archived_pages' region_slug=region.slug language_slug=language.slug %}" class="font-bold text-xs text-gray-800 block pb-2 hover:underline">
+                {% trans 'Archived pages' %}
+                <span class="inline-block rounded-full bg-integreat-500 text-gray-800 px-2 py-1 text-xs font-bold">
+                    {{ archived_count }}
+                </span>
+            </a>
+        </div>
```
Comment (timoludwig) [inline src/cms/templates/pages/page_tree.html:125]: PR #766
In this case as well, I think I'd prefer the original layout:
develop

File: src/frontend/css/style.scss [hunk 2]
```diff
@@ -71,8 +71,9 @@ nav#primary-navigation {
   background-color: #eff3f6;
 }
 
-h2.heading {
-  font-size: 1.65rem;
+h1.heading {
+  @apply text-3xl;
+  @apply mb-2;
```
Comment (timoludwig) [inline src/frontend/css/style.scss:76]:
```suggestion
  @apply text-3xl mb-2;
```
I think it should be possible to write those assignments in one line.

File: src/cms/templates/users/admin/list.html [hunk 3]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/3 flex flex-wrap">
-            <h2> 
-            {% trans 'Manage User'%}
-            </h2>
+            <h1 class="heading"> 
+            {% trans 'Manage User' %}
```
Comment (timoludwig) [inline src/cms/templates/users/admin/list.html:9]:
```suggestion
                {% trans 'Manage User' %}
```

File: src/cms/templates/settings/admin_settings.html [hunk 4]
```diff
@@ -6,7 +6,9 @@
 	<div class="header">
 		<div class="flex flex-wrap">
 			<div class="w-1/4">
-				<h2 class="heading font-normal mb-2">{% trans 'Admin Settings' %}</h2>
+				<h1 class="heading">
+				{% trans 'Admin Settings' %}
```
Comment (timoludwig) [inline src/cms/templates/settings/admin_settings.html:10]:
```suggestion
				    {% trans 'Admin Settings' %}
```

File: src/cms/templates/roles/list.html [hunk 5]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/2 flex flex-wrap">
-            <h2> 
+            <h1 class="heading">
             {% trans 'Manage Roles'%}
```
Comment (timoludwig) [inline src/cms/templates/roles/list.html:9]:
```suggestion
                {% trans 'Manage Roles'%}
```

File: src/cms/templates/regions/region_list.html [hunk 6]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/3 flex flex-wrap">
-            <h2> 
+            <h1 class="heading">
             {% trans 'Manage Regions'%}
```
Comment (timoludwig) [inline src/cms/templates/regions/region_list.html:9]:
```suggestion
                {% trans 'Manage Regions'%}
```

File: src/cms/templates/pois/poi_list.html [hunk 7]
```diff
@@ -7,32 +7,22 @@
 {% block content %}
 <div class="table-header">
     <div class="flex flex-wrap">
-        <div class="w-1/2">
-            <h2 class="heading font-normal mb-2">{% trans 'Locations on map' %}</h2>
+        <div class="w-1/3">
+            <h1 class="heading">
+            {% trans 'Locations on map' %}
```
Comment (timoludwig) [inline src/cms/templates/pois/poi_list.html:12]:
```suggestion
                {% trans 'Locations on map' %}
```

File: src/cms/templates/pages/page_tree_archived.html [hunk 8]
```diff
@@ -9,7 +9,9 @@
 {% block content %}
 <div class="table-header">
     <div class="w-1/2">
-        <h2 class="heading font-normal mb-2">{% trans 'Archived Pages' %}</h2>
+        <h1 class="heading">
+        {% trans 'Archived Pages' %}
```
Comment (timoludwig) [inline src/cms/templates/pages/page_tree_archived.html:13]:
```suggestion
            {% trans 'Archived Pages' %}
```

File: src/cms/templates/organizations/organization_list.html [hunk 9]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/2 flex flex-wrap">
-            <h2> 
+            <h1 class="heading"> 
             {% trans 'Manage Organizations'%}
```
Comment (timoludwig) [inline src/cms/templates/organizations/organization_list.html:9]:
```suggestion
                {% trans 'Manage Organizations' %}
```

File: src/cms/templates/offers/offer_list.html [hunk 10]
```diff
@@ -5,19 +5,19 @@
 {% block content %}
 <div class="table-header">
     <div class="flex flex-wrap">
-        <div class="w-1/2">
-            <h2 class="heading font-normal mb-2">{% trans 'Offers' %}</h2>
+        <div class="w-1/3">
+            <h1 class="heading">
+            {% trans 'Offers' %}
```
Comment (timoludwig) [inline src/cms/templates/offers/offer_list.html:10]:
```suggestion
                {% trans 'Offers' %}
```

File: src/cms/templates/offer_templates/offer_template_list.html [hunk 11]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/2 flex flex-wrap">
-            <h2> 
+            <h1 class="heading"> 
             {% trans 'Offer Templates'%}
```
Comment (timoludwig) [inline src/cms/templates/offer_templates/offer_template_list.html:9]:
```suggestion
                {% trans 'Offer Templates'%}
```

File: src/cms/templates/media/media_form.html [hunk 12]
```diff
@@ -2,6 +2,11 @@
 {% load i18n %}
 {% block content %}
 {% load static %}
+    <div class="w-1/3 flex flex-wrap">
+        <h1 class="heading">
+        {% trans 'Upload File'%}
```
Comment (timoludwig) [inline src/cms/templates/media/media_form.html:7]:
```suggestion
            {% trans 'Upload File' %}
```

File: src/cms/templates/languages/language_list.html [hunk 13]
```diff
@@ -5,9 +5,9 @@
 <div class="table-header">
     <div class="flex flex-wrap">
         <div class="w-1/2 flex flex-wrap">
-            <h2> 
+            <h1 class="heading">
             {% trans 'Manage Languages'%}
```
Comment (timoludwig) [inline src/cms/templates/languages/language_list.html:9]:
```suggestion
                {% trans 'Manage Languages' %}
```

File: src/cms/templates/feedback/admin_feedback_list.html [hunk 14]
```diff
@@ -8,7 +8,9 @@
 {% block content %}
 <div class="flex flex-wrap">
     <div class="w-1/2 flex">
-        <h2 class="heading font-normal mb-2">{% trans 'Technical Feedback' %}</h2>
+        <h1 class="heading">
+        {% trans 'Technical Feedback' %}
```
Comment (timoludwig) [inline src/cms/templates/feedback/admin_feedback_list.html:12]:
```suggestion
            {% trans 'Technical Feedback' %}
```

File: src/cms/templates/dashboard/dashboard.html [hunk 15]
```diff
@@ -6,7 +6,9 @@
 
 {% block content %}
     <div class="dashboard">
-        <h1 class="text-2xl mb-3">{% trans 'My Dashboard' %}</h1>
+        <h1 class="heading">
+        {% trans 'My Dashboard' %}
```
Comment (timoludwig) [inline src/cms/templates/dashboard/dashboard.html:10]:
```suggestion
            {% trans 'My Dashboard' %}
```

**Review by timoludwig — COMMENTED** [review]
In general, this looks good! :+1:

I noticed that you also tried to make the general layout of the heading in combination with the search bar and the main buttons consistent. However - I think of the two variants we have used until now, you picked the slightly suboptimal version. When the search bar is floating in the middle between the heading and the buttons, it looks less appealing to me than when it's all aligned to the left or right. Also, from a UX perspective, it makes sense to separate information (aka the heading) and actions (aka buttons and the search bar) a little bit.
So I think it would be good to revert the changes you have made to the layout :neutral_face: 
If you want to go the extra mile, you can also change all places where the search bar is floating in the middle (the region list and the admin user list) and adapt it to the aligned version, but that's not necessarily part of this task. :smile:

File: src/cms/templates/analytics/translation_coverage.html [hunk 16]
```diff
@@ -9,7 +9,9 @@
 {% block content %}
 <div class="row">
     <div class="col-sm-12">
-        <h2 class="heading font-normal mb-2">{% trans 'Translation Report' %}</h2>
+        <h1 class="heading">
+        {% trans 'Translation Report' %}
```
[the remark this answers is not in the document]
Comment (Wiedemsa) [reply src/cms/templates/analytics/translation_coverage.html:13]: which option should be preffered in terms of styling in your opinion? I found both in our code, one line and spreaded. I also found out the html documentation tells, preffering one line but not haveing to long lines. So i would choose one line in our case.

File: src/cms/templates/events/event_list.html [hunk 17]
```diff
@@ -90,6 +79,17 @@ <h2 class="heading font-normal mb-2">{% trans 'Events' %}</h2>
...
+        <div class="flex flex-wrap justify-end">
+            <a href="{% url 'events_archived' region_slug=region.slug language_slug=language.slug %}"
+               class="font-bold text-xs text-gray-800 block pb-2 hover:underline">
+                {% trans 'Archived events' %}
+                <span class="inline-block rounded-full bg-integreat-500 text-gray-800 px-2 py-1 text-xs font-bold">
+                    {{ archived_count }}
+                </span>
+            </a>
+        </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (Wiedemsa) [reply src/cms/templates/events/event_list.html:92]: You are right with this one. We had some different variations and i tried to match them all, but it didnt came in mind to seperate headings and actions buttons. I will adapt our pages to match your proposal :)

File: src/cms/templates/analytics/translation_coverage.html (hunk 16 above)
[the remark this answers is not in the document]
Comment (timoludwig) [reply src/cms/templates/analytics/translation_coverage.html:13]: At the moment, we don't have a strict formatting guide for HTML files. So I think both ways are acceptable as long as the indentation is correct. But you're right, if it fits in one line without being to long, it makes sense to put it into one line.

**Comment by Wiedemsa** [comment]
What about pages without a searchbar?
Should i keep heading and buttons in one line or make the layout consistent and separate it into two lines?
Seperated:
image

Single line:
image

imo single line looks better

**Comment by timoludwig** [comment]
> What about pages without a searchbar?
> Should i keep heading and buttons in one line or make the layout consistent and separate it into two lines?
> 
> imo single line looks better 

Yes, I agree. @ulliholtgrave do you have more insight what we should prefer from a UX perspective?

**Comment by ulliholtgrave** [comment]
I guess that's okay. Over here is a clear separation due to the left and right alignment.

File: src/cms/templates/users/region/list.html [hunk 18]
```diff
@@ -3,6 +3,9 @@
 {% block content %}
 {% load static %}
 <div class="table-header">
+    <div class="flex flex-wrap">
+        <h1 class="heading">{% trans 'All users' %}</h1>
```
Comment (timoludwig) [inline src/cms/templates/users/region/list.html:7]:
```suggestion
        <h1 class="heading">{% trans 'Users' %}</h1>
```

**Review by timoludwig — COMMENTED** [review]
Thanks! :muscle: 
Looks good now, I just have a few additional suggestions regarding some translations...

And can you rebase your branch onto develop again and resolve the conflicts in the translation file (with `./dev-tools/resolve_translation_conflicts.sh`? Thanks!

**Pull request closed without merging**